internal/core/items.rs

```diff
@@ -200,6 +200,18 @@ impl ItemWeak {
     pub fn upgrade(&self) -> Option<ItemRc> {
         self.component.upgrade().map(|c| ItemRc::new(c, self.index))
     }
+
+    pub fn component(&self) -> ComponentWeak {
+        self.component.clone()
+    }
+
+    pub fn index(&self) -> usize {
+        self.index
+    }
+
+    pub fn is_none(&self) -> bool {
+        VWeak::ptr_eq(&self.component, &Default::default())
```

Comment:
That's not correct because if the strong count reach zero, the weak becomes none but its internal pointer is not updated
I'd leave this function out.

Reply:
I basically plan to abuse `ItemWeak` as a `Option<Component, usize>` that works from C++ and rust.

I actually mostly care for the index and much less for the Component, so I could probably get by with an `Option<usize>` in most places.

Only when I start to stitch together several component-wide ItemTrees do I need to look at the component at all.

So yes, ItemWeak can be invalid in more ways than being `None`, but it is a nice way to check whether some call error-ed out or not.

Reply:
Ok, make sense.  But please document that function properly then.